### DISCUSSION TITLE
Exit codes & fail-fast config

### DISCUSSION THREAD

**Discussion by paambaati** [jurassiscripts/velociraptor#67] [Q&A]
I’m evaluating `velociraptor` for a new project and I noticed the documentation doesn’t talk about exit codes and how they’re handled.

To be more specific, let’s say I run 3 scripts, and the second fails, what happens? Is there a configuration that toggles fail-fast? Turning it on will exit after script 2 errors, and turning it off will proceed to execute script 3 too, and finally exit with the error code of script 2.

**Comment by umbopepato** [comment]
Hey @paambaati! Glad you're checking out Velociraptor 🙂

> let’s say I run 3 scripts, and the second fails, what happens?

Vr would kill all the other pending processes as soon as the second script fails (like concurrently's `--kill-others-on-fail`).

> finally exit with the error code of script 2

The exit code you'll receive in this case is specific to vr.

> Is there a configuration that toggles fail-fast?

As of now this is not configurable but I guess we could use script config files themselves to add some dedicated configuration parameters.
Also, you're right, the documentation needs some more info about this, I'll add it asap.

**Reply by paambaati** [reply]
> The exit code you'll receive in this case is specific to vr.

Wouldn't it be more "correct" to forward the original script's exit code?

**Reply by umbopepato** [reply]
Seems reasonable! The only thing to watch for is to not mistake these exit codes for vr's internal ones with the same value (I'll add this to the docs as well).
I created #68 to keep track of this, will work on it as soon as I'll have a minute.
